### PR TITLE
Improvements and new datasets

### PR DESCRIPTION
**Pull request by albanPi**
## Modifications
- Added a pipeline in `gridsearch_solver.py` with OneHotEncoder.
- Added new metrics (balanced accuracy and ROC AUC).
- Modified the `parameter_grid` names of the solvers to make it work with the pipeline. Also added to the existing dataset a categorical indicator to satisfy the OneHotEncoder process.
- Added new datasets from [Leo's paper](https://hal.science/hal-03723551v2) and package requirements for these datasets.

### CODE REVIEW

File: datasets/KDDCup09_upselling.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    from sklearn.model_selection import train_test_split
+    import openml
+
+
+class Dataset(BaseDataset):
+
+    name = "KDDCup09_upselling"
+
+    install_cmd = "conda"
+    requirements = ["pip:openml", "pip:chardet"]
```
Comment (tomMoral) [inline datasets/KDDCup09_upselling.py:14]: pourquoi tu as besoin de `chardet`?

File: datasets/compass.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    requirements = ["pip:openml", "pip:chardet"]
+
+    parameters = {
+        'test_size': [0.25],
+        'seed': [42]
+    }
+
+    def get_data(self):
+        rng = np.random.RandomState(self.seed)
+        dataset = openml.datasets.get_dataset(44162)
```
Comment (tomMoral) [inline datasets/compass.py:23]: Ça fait énormément de code dupliqué alors que c'est toujours la meme chose sauf cette ligne
Est ce que tu pourrais faire un truc avec un dictionaire:

```python
DATASETS = {
    'california': XXX,
    'compass': 44162,
   ...
}
```

puis faire un dataset avec un parametre `name = ['california', 'compass', ...]` et selectionné le bon dataset? comme ça tu n'aurait qu'un fichier.

(maybe better: `name = list(DATASET.keys())`)

File: solvers/xgboost.py [hunk 3]
```diff
@@ -1,20 +1,20 @@
...
+
+class Solver(GSSolver):
+
+    name = 'XGBoost'
+
+    requirements = ['py-xgboost']
+
+    parameter_grid = {
+        'model__n_estimators': [100, 1000, 2000],
+        'model__max_depth': range(1, 12)
```
Comment (tomMoral) [inline solvers/xgboost.py:16]: I would not modify this like this but rather modify in `grid_search_solver` the parameter grid to add `model__` to the keys.

File: datasets/KDDCup09_upselling.py (hunk 1 above)
Comment (albanPi) [reply datasets/KDDCup09_upselling.py:14]: Car `openml` me demande d'installer `chardet` pour pouvoir importer les datasets

File: datasets/compass.py (hunk 2 above)
Comment (albanPi) [reply datasets/compass.py:23]: C'est plus intelligent en effet. Je vais changer ça et voir comment bien l'implémenter.

File: solvers/xgboost.py (hunk 3 above)
Comment (albanPi) [reply solvers/xgboost.py:16]: This will change with PR #7 (once updated)
